From 7aeebb61145d3a2082058102bdf33634cce21c4b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 16 Nov 2016 17:32:40 +0000
Subject: [PATCH 1/5] daemon: Export the "action-icons" capability to clients

The freedesktop specification states that clients may opt to use named icons
instead of textual labels on action buttons. In this instance, the icon name
is passed as the 'key' (actions[i+1]) element in the actions data passed
to the daemon.

For themes to render the icon, they should check the hints passed by the
client for a boolean value of "action-icons", and if it is set they should
then create an image button based on the key, and not the label.

Many applications now make use of action-icons, such as media players, to
provide self describing actions with well known icon names, such as playback
controls. In the absence of support for action-icons they fallback to less
useful buttons and generally less actions.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/daemon/daemon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/daemon/daemon.c b/src/daemon/daemon.c
index eb2fe9b..149025a 100644
--- a/src/daemon/daemon.c
+++ b/src/daemon/daemon.c
@@ -1610,6 +1610,7 @@ gboolean notify_daemon_get_capabilities(NotifyDaemon* daemon, char*** caps)
 {
 	GPtrArray* a = g_ptr_array_new ();
 	g_ptr_array_add (a, g_strdup ("actions"));
+	g_ptr_array_add (a, g_strdup ("action-icons"));
 	g_ptr_array_add (a, g_strdup ("body"));
 	g_ptr_array_add (a, g_strdup ("body-hyperlinks"));
 	g_ptr_array_add (a, g_strdup ("body-markup"));
-- 
2.10.2

